src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            is_linked = self.server.link(state)
+            return is_linked
+
+    def is_linked(self):
+        """
+        Returns: (bool) True if Z follows free working distance.
+        When Z follows FWD and Z-axis of stage moves, FWD is updated to keep image in focus.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.is_linked()
+
+#### CHAMBER CONTROL
+
+    def pump(self):
+        """Pump the microscope's chamber. Note that pumping takes some time. This is blocking."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            # Pump/vent functions can take a long time, so change timeout
+            self.server._pyroTimeout = 300  # seconds
+            try:
+                self.server.pump()
+            except TimeoutError:
+                logging.warning("Pumping timed out after %s s. Check the xt user interface for the current status " +
+                                "of the chamber", self.server._pyroTimeout)
+            finally:
+                self.server._pyroTimeout = 30  # seconds
+
+    def vent(self):
+        """Vent the microscope's chamber. Note that venting takes time (appr. 3 minutes). This is blocking."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            # Pump/vent functions can take a long time, so change timeout
+            self.server._pyroTimeout = 300  # seconds
+            try:
+                self.server.vent()
+            except TimeoutError:
+                logging.warning("Venting timed out after %s s. Check the xt user interface for the current status " +
+                                "of the chamber", self.server._pyroTimeout)
+            finally:
+                self.server._pyroTimeout = 30  # seconds
+
+    def get_chamber_state(self):
+        """Returns: (str) the vacuum state of the microscope chamber to see if it is pumped or vented,
+        possible states: "vacuum", "vented", "prevac", "pumping", "venting","vacuum_error" """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_chamber_state()
+
+
+    def get_pressure(self):
+        """Returns: (float) the chamber pressure in pascal, or -1 in case the system is vented."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            pressure = self.server.get_pressure()
+            return pressure
+
+    def pressure_info(self):
+        """Returns (dict): the unit and range of the pressure."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.pressure_info()
+
+##### BEAM CONTROL
+
+    def set_scan_mode(self, mode: str, channel: str, value: Union[float, dict] = None) -> None:
```

Comment:
```suggestion
    def set_scan_mode(self, mode: str, channel: str, value: Union[float, dict]) -> None:
```

Reply:
Not every scan mode takes a value, so changing it to optional arg